app/post.go

```diff
@@ -941,9 +920,9 @@ func (a *App) DoPostAction(postId, actionId, userId, selectedOption string) *mod
 		return model.NewAppError("DoPostAction", "api.post.do_action.action_integration.app_error", nil, fmt.Sprintf("status=%v", resp.StatusCode), http.StatusBadRequest)
 	}
 
-	var response model.PostActionIntegrationResponse
-	if err := json.NewDecoder(resp.Body).Decode(&response); err != nil {
-		return model.NewAppError("DoPostAction", "api.post.do_action.action_integration.app_error", nil, "err="+err.Error(), http.StatusBadRequest)
+	response := model.PostActionIntegrationResponseFromJson(resp.Body)
```

Comment:
I thought it would be best to use a method we already have instead of writing new code. If this don't belong in this PR, I will just revert this change.

Reply:
I think it might be best to revert this one as it does reduce the level of information in the error if it fails.

Reply:
Okay